backend/dataall/core/permissions/services/tenant_policy_service.py

```diff
@@ -25,9 +18,130 @@ def is_tenant_admin(groups: [str]):
 
         return False
 
+    @staticmethod
+    def validate_admin_access(username, groups, action):
+        if not TenantPolicyValidationService.is_tenant_admin(groups):
+            raise exceptions.UnauthorizedOperation(
+                action=action,
+                message=f'User: {username} is not allowed to manage tenant permissions',
+            )
+
+    @staticmethod
+    def validate_find_tenant_policy(group_uri, tenant_name):
+        if not group_uri:
+            raise exceptions.RequiredParameter(param_name='group_uri')
+        if not tenant_name:
+            raise exceptions.RequiredParameter(param_name='tenant_name')
+
+    @staticmethod
+    def validate_attach_tenant_policy(group, permissions, tenant_name):
+        if not group:
+            raise exceptions.RequiredParameter(param_name='group')
+        if not permissions:
+            raise exceptions.RequiredParameter(param_name='permissions')
+        if not tenant_name:
+            raise exceptions.RequiredParameter(param_name='tenant_name')
+
+    @staticmethod
+    def validate_save_tenant_policy(group, tenant_name):
+        if not group:
+            raise exceptions.RequiredParameter(param_name='group')
+        if not tenant_name:
+            raise exceptions.RequiredParameter(param_name='tenant_name')
+
+    @staticmethod
+    def validate_add_permission_to_tenant_policy_params(group, permissions, policy, tenant_name):
+        if not group:
+            raise exceptions.RequiredParameter(param_name='group')
+        TenantPolicyValidationService.validate_add_permissions_params(permissions, policy, tenant_name)
+
+    @staticmethod
+    def validate_add_permissions_params(permissions, policy, tenant_name):
```

Comment:
TenantPolicy.TENANT_NAME is only used in `dataall/core/permissions/services/tenant_policy_service.py` and seems more of a TenantPolicyService variable than a `dataall/core/permissions/db/tenant/tenant_policy_repositories.py` constant. I think we could move it to the service